### PR TITLE
misskey: fix darwin build

### PR DESCRIPTION
**Pull request by ofalvai**
## Description of changes

Hydra:

https://hydra.nixos.org/job/nixpkgs/trunk/misskey.aarch64-darwin
https://hydra.nixos.org/job/nixpkgs/trunk/misskey.x86_64-darwin

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ofalvai** [comment]
Result of `nixpkgs-review pr 336087` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>misskey</li>
  </ul>
</details>

**Comment by ofalvai** [comment]
Well, it's not fully fixing x86_64, I get a different error:

```
misskey-x86_64-darwin> /nix/store/p89f52rwjz8h9ik440z25nhwj67210jq-libcxx-16.0.6-dev/include/c++/v1/new:355:14: error: no member named 'aligned_alloc' in the global namespace
misskey-x86_64-darwin>     return ::aligned_alloc(__alignment, __size > __rounded_size ? __size : __rounded_size);
misskey-x86_64-darwin>            ~~^
misskey-x86_64-darwin> 1 error generated.
```

I believe this is because the x86_64 darwin stdenv is stuck at the 10.12 SDK, although [it's changing soon](https://discourse.nixos.org/t/on-the-future-of-darwin-sdks-or-how-you-can-stop-worrying-and-put-the-sdk-in-build-inputs/50574), so I'd rather merge this for fixing at least aarch64-darwin

**Comment by ofalvai** [comment]
Result of `nixpkgs-review pr 336087` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>misskey</li>
  </ul>
</details>

File: pkgs/by-name/mi/misskey/package.nix [hunk 1]
```diff
@@ -27,6 +28,8 @@ stdenv.mkDerivation (finalAttrs: {
     fetchSubmodules = true;
   };
 
+  buildInputs = lib.optionals stdenv.isDarwin [ xcbuild.xcrun ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/mi/misskey/package.nix:31]: Should this go into nativeBuildInputs?

Comment (ofalvai) [reply pkgs/by-name/mi/misskey/package.nix]: Thank you, fixed

**Pull request merged**